### ISSUE TITLE
ServiceEntries targetPort change is not getting updated

### ISSUE THREAD

**Issue by rvennam** [istio/istio#37022]
### Bug Description

```
cat <<EOF | kubectl apply -f -
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: se-update
spec:
  hosts:
  - reviews.global
  location: MESH_INTERNAL
  ports:
  - name: http
    number: 9080
    protocol: http
    targetPort: 9081 # NOTE! <---
  resolution: STATIC
  workloadSelector:
    labels:
      app: reviews
EOF
```
```
istioctl pc endpoints deploy/productpage-v1 | grep reviews.global
10.44.0.49:9081                  HEALTHY     OK                outbound|9080||reviews.global
10.44.1.55:9081                  HEALTHY     OK                outbound|9080||reviews.global
10.44.1.56:9081                  HEALTHY     OK                outbound|9080||reviews.global
```
Update the targetPort and apply again:
```
cat <<EOF | kubectl apply -f -
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: se-update
spec:
  hosts:
  - reviews.global
  location: MESH_INTERNAL
  ports:
  - name: http
    number: 9080
    protocol: http
    targetPort: 9080 # CHANGE TO 9080
  resolution: STATIC
  workloadSelector:
    labels:
      app: reviews
EOF
```
```
serviceentry.networking.istio.io/se-update configured
~ 🏎  $ istioctl pc endpoints deploy/productpage-v1 | grep reviews.global
10.44.0.49:9081                  HEALTHY     OK                outbound|9080||reviews.global
10.44.1.55:9081                  HEALTHY     OK                outbound|9080||reviews.global
10.44.1.56:9081                  HEALTHY     OK                outbound|9080||reviews.global
```

Endpoints still have the old 9081 value. `istioctl ps` shows all SYNCED

### Version

```prose
$ istioctl version
client version: 1.11.5
control plane version: 1.11.5
data plane version: 1.11.5 (10 proxies)
```

**Comment by howardjohn** [comment]
I vaguely recall some similar PRs lately. Would be good for you or someone on our side to check if this is still happening on `master`

**Comment by hzxuzhonghu** [comment]
Yes, we changed something recently.

**Comment by linsun** [comment]
I tried it on the latest master, still not seeing the port updated.  Not seeing any push related to the SE update in istiod log either.

```
vagrant@impish:~$ cat <<EOF | kubectl apply -f -
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: se-update
spec:
  hosts:
  - reviews.global
  location: MESH_INTERNAL
  ports:
  - name: http
    number: 9080
    protocol: http
    targetPort: 9080 # CHANGE TO 9080
  resolution: STATIC
  workloadSelector:
    labels:
      app: reviews
EOF
serviceentry.networking.istio.io/se-update configured
vagrant@impish:~$ istioctl pc endpoints deploy/productpage-v1 | grep reviews.global
10.244.0.10:9081                 HEALTHY     OK                outbound|9080||reviews.global
10.244.0.12:9081                 HEALTHY     OK                outbound|9080||reviews.global
10.244.0.9:9081                  HEALTHY     OK                outbound|9080||reviews.global
vagrant@impish:~$ istioctl ps
NAME                                                   CDS        LDS        EDS        RDS          ISTIOD                     VERSION
details-v1-79f774bdb9-bjdzv.default                    SYNCED     SYNCED     SYNCED     SYNCED       istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
istio-ingressgateway-6b86bf4858-vq7p6.istio-system     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
productpage-v1-6b746f74dc-995pt.default                SYNCED     SYNCED     SYNCED     SYNCED       istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
ratings-v1-b6994bb9-csfzr.default                      SYNCED     SYNCED     SYNCED     SYNCED       istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
reviews-v1-545db77b95-8f99s.default                    SYNCED     SYNCED     SYNCED     SYNCED       istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
reviews-v2-7bf8c9648f-mpgts.default                    SYNCED     SYNCED     SYNCED     SYNCED       istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
reviews-v3-84779c7bbc-vq4tx.default                    SYNCED     SYNCED     SYNCED     SYNCED       istiod-c55549fb4-4bbg9     1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
vagrant@impish:~$ istioctl pc endpoints deploy/productpage-v1 | grep reviews.global
10.244.0.10:9081                 HEALTHY     OK                outbound|9080||reviews.global
10.244.0.12:9081                 HEALTHY     OK                outbound|9080||reviews.global
10.244.0.9:9081                  HEALTHY     OK                outbound|9080||reviews.global
vagrant@impish:~$ istioctl version
client version: 1.12.1
control plane version: 1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7
data plane version: 1.14-alpha.c8b803876bc0f5b2a2b27c570283e0f9615ec0d7 (7 proxies)
```

**Comment by linsun** [comment]
After ~20 mins, i started to see more push requests in my istiod log and the port number is updated.

**Comment by hzxuzhonghu** [comment]
THis is because when we convert SE to service, the target port is not used at all, so the handler treats it as no update at all, so no xds triggered.

**Comment by dddddai** [comment]
I can't reproduce this on master, not sure if I missed something?
```sh
[root@new ~]# istioctl version
client version: 1.10.3
control plane version: unknown   # it's master
data plane version: 1.10.3 (7 proxies)
```
```sh
[root@new ~]# istioctl ps
NAME                                                   CDS        LDS        EDS        RDS        ISTIOD                      VERSION
details-v1-66b6955995-22gc5.default                    SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
istio-ingressgateway-75f95745cc-j7rj2.istio-system     SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
productpage-v1-5d9b4c9849-tmf5g.default                SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
ratings-v1-fd78f799f-4pngm.default                     SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
reviews-v1-6549ddccc5-4dsnc.default                    SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
reviews-v2-76c4865449-bdlgk.default                    SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
reviews-v3-6b554c875-hj2qw.default                     SYNCED     SYNCED     SYNCED     SYNCED     istiod-6bb6dcdb4d-lfqtj     1.10.3
```
```sh
[root@new ~]# istioctl pc endpoints deploy/productpage-v1 | grep reviews.global
10.244.1.14:9080                 HEALTHY     OK                outbound|9080||reviews.global
10.244.1.17:9080                 HEALTHY     OK                outbound|9080||reviews.global
10.244.1.18:9080                 HEALTHY     OK                outbound|9080||reviews.global
```
```sh
[root@new ~]# cat <<EOF | kubectl apply -f -
> apiVersion: networking.istio.io/v1alpha3
> kind: ServiceEntry
> metadata:
>   name: se-update
> spec:
>   hosts:
>   - reviews.global
>   location: MESH_INTERNAL
>   ports:
>   - name: http
>     number: 9080
>     protocol: http
>     targetPort: 9081 # NOTE! <---
>   resolution: STATIC
>   workloadSelector:
>     labels:
>       app: reviews
> EOF
serviceentry.networking.istio.io/se-update configured
[root@new ~]# istioctl pc endpoints deploy/productpage-v1 | grep reviews.global
10.244.1.14:9081                 HEALTHY     OK                outbound|9080||reviews.global
10.244.1.17:9081                 HEALTHY     OK                outbound|9080||reviews.global
10.244.1.18:9081                 HEALTHY     OK                outbound|9080||reviews.global
```

**Comment by linsun** [comment]
@GregHanson I think you also tried this a few weeks ago and said it is fixed in master?  Do you know if it is fixed in 1.13.x?

**Comment by GregHanson** [comment]
This is resolved in Istio 1.13.3
